src/NumberInput.jsx

```diff
@@ -56,6 +63,23 @@ export default React.createClass({
       this.getDefaultState(nextProps))
   },
 
+  componentWillUpdate(nextProps, nextState) {
+    if (canUseDOM) {
+      // Check if we need to restore text selection after pending update.
+      let node = ReactDOM.findDOMNode(this.refs.input);
+      this._selectAll = document.activeElement === node
+        && this.state.stringValue !== nextState.stringValue
```

Comment:
👋  hey there. I like the idea of this but i'm a bit worry about trying to mimic the browser behavior here. One concern is that this will maintain the selection at times that aren't the user tabbing into field, such as pasting.